src/Compilers/VisualBasic/Portable/Compilation/VisualBasicDiagnosticFilter.vb

```diff
@@ -86,8 +96,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             ' If location is available, check warning directive state.
             If location IsNot Nothing AndAlso location.SourceTree IsNot Nothing AndAlso
-           location.SourceTree.GetWarningState(id, location.SourceSpan.Start) = ReportDiagnostic.Suppress Then
-                Return ReportDiagnostic.Suppress
+                location.SourceTree.GetWarningState(id, location.SourceSpan.Start) = ReportDiagnostic.Suppress Then
+                hasDisableDirectiveSuppression = True
```

Comment:
Please rename to "hasSourceSuppression" in both languages as we had discussed before.